### ISSUE TITLE
sanity check in PasswordInspector::smart_binsearch() fails when processing small files

### ISSUE THREAD

**Issue by ola-ct** [ola-ct/pwned#5]
Write a test to cover smart searches in small files, e.g. in pwned-lib/test/testset-10000-existent-collection1+2+3+4+5.md5.

Proposal to extend pwned-lib/test/test_inspector.cpp:

```cpp
BOOST_AUTO_TEST_CASE(test_existent_smart)
{
  const std::string inputFilename = "../../../../pwned-lib/test/testset-10000-existent-collection1+2+3+4+5.md5";
  const uint64_t size = boost::filesystem::file_size(inputFilename);
  const uint64_t hashCount = size / pwned::PHC::size;
  std::vector<pwned::PHC> phcs;
  phcs.reserve(hashCount);
  std::ifstream testset(inputFilename, std::ios::binary);
  pwned::PasswordInspector inspector(inputFilename);
  pwned::PHC phc;
  uint64_t nFound = 0;
  while (phc.read(testset))
  {
    if (inspector.smart_binsearch(phc.hash).count > 0)
    {
      ++nFound;
    }
  }
  BOOST_TEST(nFound == hashCount);
}

BOOST_AUTO_TEST_CASE(test_nonexistent_smart)
{
  const std::string inputFilename = "../../../../pwned-lib/test/testset-10000-existent-collection1+2+3+4+5.md5";
  const std::string nonExistentInputFilename = "../../../../pwned-lib/test/testset-10000-nonexistent-collection1+2+3+4+5.md5";
  const uint64_t size = boost::filesystem::file_size(nonExistentInputFilename);
  const uint64_t hashCount = size / pwned::PHC::size;
  std::vector<pwned::PHC> phcs;
  phcs.reserve(hashCount);
  std::ifstream testset(nonExistentInputFilename, std::ios::binary);
  pwned::PasswordInspector inspector(inputFilename);
  pwned::PHC phc;
  uint64_t nNotFound = 0;
  while (phc.read(testset))
  {
    if (inspector.smart_binsearch(phc.hash).count == 0)
    {
      ++nNotFound;
    }
  }
  BOOST_TEST(nNotFound == hashCount);
}
```

**Comment by ola-ct** [comment]
Failure can now easily be reproduced with `make test` (see fa13ee173ebcb31ef546804caf4f3e3afbb59b5f).

**Comment by ola-ct** [comment]
Fixed with c46df9b22776736376a734432fbd356d7dccb6c1.
